flink-core/src/main/java/org/apache/flink/api/common/functions/RuntimeContext.java

```diff
@@ -65,6 +65,22 @@
 	int getIndexOfThisSubtask();
 
 	/**
+	 * Gets the attempt number of this parallel subtask. First attempt is numbered 0.
```

Comment:
I would add here something like: "The attempt number corresponds to how many times the task has been restarted (after a failure or cancellation) since the job was initially started."